ivy/functional/ivy/general.py

```diff
@@ -1497,6 +1497,17 @@ def cumprod(
     -------
     ret
         Input array with cumulatively multiplied elements along axis.
+        
+    Examples
+    --------
+    >>> x = ([a, b, c])
```

Comment:
I think the type of array should be made clear `ivy.Array()` or `ivy.NativeArray`

Reply:
Since the function is calling backend specific methods`_cur_backend(x).cumprod(x, axis, exclusive, out=out)`, I think it should take `ivy.NativeArray` as well. @saeedashrraf , what do you think?

Reply:
I might be wrong, but my process of thought, was that if the cumprod function is native, even if it calls a backend method, it will return an ivy.Array(). This I saw in the descriptions of the two methods (ivy.Array() and ivy.NativeArray()).
But I share the opinion that the function can take both. 
What would be the proposition? do you want me to alter the method of arranging the array in the examples, or do you prefer if I clarify in the section parameters, that you can use both methods (this part was merged 2 months ago)
I thought that it was implicit in the definition of parameters, but there is no harm in clarifying the subject even more.
I am at your disposal.

Reply:
all cleared reading the guide, chapter 11, you were right! added this example, and a container one, with the modifications requested in the review!